uxarray/grid.py

```diff
@@ -141,6 +151,60 @@ def __init_grid_var_attrs__(self) -> None:
                 if value in self.ds.dims:
                     setattr(self, key, len(self.ds[value]))
 
+    def __from_multi_faces_vert__(self, dataset):
+        """Create a grid with multiple faces with vertices specified by the
+        given argument. Called by :func:`__init__`.
+
+        Parameters
+        ----------
+        dataset : ndarray
+            ndarry of vertex coordinates that form multiple faces.
+        """
```

Comment:
If this implementation works works for an arbitrary number of faces (including just a single one), we should patch up `__from_vert__` instead of creating a separate function.

Reply:
This one will work for a single one (as illustrated in the `test__from_multi_faces_vert__`)

The reason why I created a separate function for multiple test cases is that I noticed the input for `Grid.__from_vers__()` is 

```python
np.array([[0, 0], [2, 0], [0, 2], [2, 2]])
```

Adding another function to handle the `multiple faces` will allow users to initialize a single face through

```python
np.array([[0, 0], [2, 0], [0, 2], [2, 2]])
```

or 

```python
np.array([
            np.array([[150, 10], [160, 20], [150, 30], [135, 30], [125, 20],
                      [135, 10]])])
```

But either patching and separating works for me, as long as we can initialized the`Grid` object through multiple face objects

Reply:
I am favoring to expand the `__from_vert__ ` function's capability to handle both single-face and multi-face grid creation instead of separate functions; is that fine @rajeeja ?

Reply:
Absolutely.

Reply:
@hongyuchen1030 Could you please make changes to get it done as an expansion to the existing `__from_vert__` as suggested? Thanks very much for the added functionality!

Reply:
> @hongyuchen1030 Could you please make changes to get it done as an expansion to the existing `__from_vert__` as suggested? Thanks very much for the added functionality!

Working on it